api/apps/api/src/modules/scenarios/project-checker.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@nestjs/common';
+import { In } from 'typeorm';
+
+import { API_EVENT_KINDS } from '@marxan/api-events';
+import { ApiEventsService } from '@marxan-api/modules/api-events';
+
+@Injectable()
+export class ProjectChecker {
+  constructor(private readonly apiEvents: ApiEventsService) {}
+
+  async isProjectReady(projectId: string): Promise<boolean> {
```

Comment:
@hotzevzl we should wait for all jobs related to a project, shouldn't we? (likewise, for setting custom project area / grid)

Reply:
sorry for having missed this - yes, for the purpose of blocking scenario operations, `isProjectReady()` should ensure that all the jobs are done, to prevent scenarios from being created while project-level setup is not complete yet (or has failed).